src/arrayMethodJoin.js

```diff
@@ -6,6 +6,32 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+
+    let separate = separator;
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (separator === undefined) {
```

Comment:
U don't need this condition, because u can set default value in parameters. Like this: function (separator = 0)